activesupport/lib/active_support/message_verifier.rb

```diff
@@ -44,7 +44,7 @@ def initialize(secret, options = {})
     #   tampered_message = signed_message.chop # editing the message invalidates the signature
     #   verifier.valid_message?(tampered_message) # => false
     def valid_message?(signed_message)
-      return if signed_message.blank?
+      return if signed_message.nil? || !signed_message.valid_encoding? || signed_message.blank?
```

Comment:
Good point, @matthewd, but we do test for nil input: https://github.com/repinel/rails/blob/fix-message-verifier-encoding-issue/activesupport/test/message_verifier_test.rb#L26